test/parallel/test-web-locks.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+// Flags: --expose-gc
+
+require('../common');
+const { describe, it } = require('node:test');
+const assert = require('node:assert');
+const { Worker } = require('worker_threads');
+
+describe('Web Locks with worker threads', () => {
+  it('should handle exclusive locks', async () => {
+    const worker = new Worker(`
+      const { parentPort } = require('worker_threads');
+      const assert = require('node:assert');
+      
+      navigator.locks.request('exclusive-test', async (lock) => {
+        assert.strictEqual(lock.mode, 'exclusive');
+        parentPort.postMessage({ success: true });
+      }).catch(err => parentPort.postMessage({ error: err.message }));
+    `, { eval: true });
+
+    const result = await new Promise((resolve) => {
+      worker.once('message', resolve);
+    });
+
+    assert.strictEqual(result.success, true);
+    await worker.terminate();
+
+    await navigator.locks.request('exclusive-test', async (lock) => {
+      assert.strictEqual(lock.mode, 'exclusive');
+      assert.strictEqual(lock.name, 'exclusive-test');
+    });
+  });
+
+  it('should handle shared locks', async () => {
+    const worker = new Worker(`
+      const { parentPort } = require('worker_threads');
+      const assert = require('node:assert');
+      
+      navigator.locks.request('shared-test', { mode: 'shared' }, async (lock) => {
+        assert.strictEqual(lock.mode, 'shared');
+        parentPort.postMessage({ success: true });
+      }).catch(err => parentPort.postMessage({ error: err.message }));
+    `, { eval: true });
+
+    const result = await new Promise((resolve) => {
+      worker.once('message', resolve);
+    });
+    assert.strictEqual(result.success, true);
+
+    await navigator.locks.request('shared-test', { mode: 'shared' }, async (lock1) => {
+      await navigator.locks.request('shared-test', { mode: 'shared' }, async (lock2) => {
+        assert.strictEqual(lock1.mode, 'shared');
+        assert.strictEqual(lock2.mode, 'shared');
+      });
+    });
+
+    await worker.terminate();
```

Comment:
There are a lot of tests here, which is excellent, but it makes it difficult to see if every case is covered. It would be good, if there isn't one already, to have a test that launches a worker thread that initiates a lock request but then have the worker get terminated/destroyed while the request is still pending to ensure that there aren't any UAFs.